### PR TITLE
[Improve][Api] Supports simultaneous config of conditions and optional options

### PR DESCRIPTION
**Pull request by corgy-w**
### Purpose of this pull request

conditional does not conflict with optional options

ex: api-path
image

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by liugddx — DISMISSED** [review]
+1

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java [hunk 1]
```diff
@@ -242,12 +242,21 @@ private void verifyDuplicateWithOptionOptions(
         }
 
         private void verifyRequiredOptionDuplicate(@NonNull RequiredOption requiredOption) {
+            verifyRequiredOptionDuplicate(requiredOption, false);
+        }
+
+        private void verifyRequiredOptionDuplicate(
+                @NonNull RequiredOption requiredOption,
+                @NonNull Boolean ignoreVerifyDuplicateWithOptionOptions) {
```
Comment (Hisoka-X) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java]:
```suggestion
                @NonNull Boolean ignoreVerifyDuplicateOptions) {
```

Comment (Hisoka-X) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java]: Could you add some comment about when should set ignore to `false` or `true`?

Comment (corgy-w) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java]: get

Comment (corgy-w) [reply seatunnel-api/src/main/java/org/apache/seatunnel/api/configuration/util/OptionRule.java]: Get, wait for add comments

**Review by Hisoka-X — APPROVED** [review]

**Comment by Hisoka-X** [comment]
Thanks @corgy-w

**Review by liugddx — APPROVED** [review]
+1

**Pull request merged**